apps/prairielearn/src/sync/fromDisk/assessments.ts

```diff
@@ -101,44 +106,35 @@ function getParamsForAssessment(
 
   let alternativeGroupNumber = 0;
   let assessmentQuestionNumber = 0;
-  const allRoleNames = (assessment.groupRoles ?? []).map((role) => role.name);
-  const assessmentCanView = assessment?.canView ?? allRoleNames;
-  const assessmentCanSubmit = assessment?.canSubmit ?? allRoleNames;
-  const alternativeGroups = (assessment.zones ?? []).map((zone) => {
+  const allRoleNames = assessment.groupRoles.map((role) => role.name);
+  const assessmentCanView = assessment.canView.length > 0 ? assessment.canView : allRoleNames;
```

Comment:
These are now defaulted to an empty array, so we should check on the array length.